### ISSUE TITLE
QA Test: Impact of accessibility changes for mobile devices

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/QA#197]
We have been doing work to make a11y features compatible with mobile devices, particularly VoiceOver. This changes the way the accessible PDOM elements are positioned, which could have large impacts on sims and performance. I have been testing mostly on an iPad2, Windows Chrome, and then spot checking on a few other platforms here and there. Things generally seem OK, but we should do more rigorous testing before merging with master. Here are three sims to test and compare.

**Before the change:**
 - [Balloons and Static Electricity](https://phet-dev.colorado.edu/html/balloons-and-static-electricity/1.5.0-pdomTransform.1/phet/balloons-and-static-electricity_en_phet.html)
 - [Gravity and Orbits](https://phet-dev.colorado.edu/html/gravity-and-orbits/1.2.0-pdomTransform.1/phet/gravity-and-orbits_en_phet.html?accessibility)
 -  [Energy Skate Park](https://phet-dev.colorado.edu/html/energy-skate-park-basics/1.4.0-pdomTransform.1/phet/energy-skate-park-basics_all_phet.html?accessibility)

**After the change:**
 - [Balloons and Static Electricity](https://phet-dev.colorado.edu/html/balloons-and-static-electricity/1.5.0-pdomTransform.1/phet/balloons-and-static-electricity_en_phet.html?mobileA11yTest)
 - [Gravity and Orbits](https://phet-dev.colorado.edu/html/gravity-and-orbits/1.2.0-pdomTransform.1/phet/gravity-and-orbits_en_phet.html?accessibility&mobileA11yTest)
 -  [Energy Skat Park: Basics](https://phet-dev.colorado.edu/html/energy-skate-park-basics/1.4.0-pdomTransform.1/phet/energy-skate-park-basics_all_phet.html?accessibility&mobileA11yTest)

Please test the behavior of each sim before and after the change, and note any differences you discover. Performance is the biggest thing that may stand out. But this changes the web document layout, so other things may have broken too. For example, please make sure that the cursor still updates correctly, nothing in the simulation looks shifted out of place, and there are no graphical artifacts in the sim.

If possible, please do comparison testing on all of our supported platforms. I know this is a big request, if that doesn't work we can decide on a few to target. Please let me know.

@ariel-phet could you please determine the priority for this task?

Pinging @emily-phet and @terracoda so they are aware of this request.

**Comment by KatieWoe** [comment]
Making a list of supported platforms to test:
- [ ] Win 10 chrome
- [ ] Win 10 firefox
- [ ] Win 10 edge
---
- [ ] Win 8.1 chrome
- [ ] Win 8.1 firefox
- [ ] Win 8.1 IE
---
- [ ] Win 7 chrome
- [ ] Win 7 firefox
- [ ] Win 7 IE
---
- [ ] Mac OS 10.14 chrome
- [ ] Mac OS 10.14 firefox
- [ ] Mac OS 10.14 safari
---
- [ ] Mac OS 10.13 chrome
- [ ] Mac OS 10.13 firefox
- [ ] Mac OS 10.13 safari
---
- [ ] Mac OS 10.12 chrome
- [ ] Mac OS 10.12 firefox
- [ ] Mac OS 10.12 safari
---
- [ ] Mac OS 10.11 chrome
- [ ] Mac OS 10.11 firefox
- [ ] Mac OS 10.11 safari
---
- [ ] Mac OS 10.10 safari
---
- [ ] Chrome OS
---
- [ ] iOS 9
- [ ] iOS 10
- [ ] iOS 11
- [ ] iOS 12

**Comment by jessegreenberg** [comment]
To clarify, this testing should be done without a screen reader enabled, we want to determine if there is an observable impact to non-a11y sim features.

**Comment by KatieWoe** [comment]
Win 10 edge performance on BASE drops fairly significantly.  Seen most when using mouse to drag balloon.

**Comment by KatieWoe** [comment]
Mac OS 10.14 Chrome Gravity and Orbits The sim slows to the point of being unusable.  It takes 10-20 seconds to respond to input and the progression of movement of the objects seems to go frame by frame.  This was not the case before the change.

**Comment by jessegreenberg** [comment]
This is very helpful @KatieWoe, thanks. For each of these cases can you comment on how long it took for the sim to become unusable or have poor performance? Is it immediately apparent or after a few seconds/minutes of running?

**Comment by KatieWoe** [comment]
So far they have all been either immediately apparent or happened very shortly after starting. @jessegreenberg

**Comment by jessegreenberg** [comment]
OK, thanks.

**Comment by lmulhall-phet** [comment]
**Windows 8 Comments**
- Chrome
  - BASE: good
  - GAO: good
  - ESPB: performance slightly worse (immediate)
- FF
  - BASE: good
  - GAO: good
  - ESPB: performance slightly worse (immediate)
- IE
  - BASE: performance worse (immediate)
  - GAO: performance worse (immediate)
  - ESPB: performance barely acceptable (immediate)
--- 
**Windows 7 Comments**
- Chrome
  - BASE: good
  - GAO: performance is garbage, the sim takes ~1 min to process a click, Chrome doesn't respond
  - ESPB: performance slightly worse (after a few minutes)
- FF
  - BASE: good
  - GAO: good
  - ESPB: good
- IE
  - BASE: good
  - GAO: good
  - ESPB: good
---
**macOS 10.12 Comments**
- Chrome
  - BASE: good
  - GAO: good
  - ESPB: good
- FF
  - BASE: good
  - GAO: good
  - ESPB: good
- Safari
  - BASE: good
  - GAO: good
  - ESPB: good
---
**macOS 10.11 Comments**
- Chrome
  - BASE: good
  - GAO: good
  - ESPB: good
- FF
  - BASE: good
  - GAO: good
  - ESPB: good
- Safari
  - BASE: good
  - GAO: good
  - ESPB: good
---
**macOS 10.10 Comments**
- Safari
  - BASE: good
  - GAO: good
  - ESPB: good
---
**Chrome OS Comments**
- Chrome
  - BASE: good
  - GAO: performance much worse (immediate)
  - ESPB: good
---
**iOS 9 Comments**
- Safari
  - BASE: good
  - GAO: performance slightly worse (immediate)
  - ESPB: good

**Comment by KatieWoe** [comment]
iOS 10:
BASE - performance actually seems a bit better after
Gravity and Orbits - Sim crashes and reloads immediately, often not even reaching the home screen.  Sim can't be used.
ESPB - ok

**Comment by KatieWoe** [comment]
iOS 11 everything looks ok.  QA is done @jessegreenberg

**Comment by jessegreenberg** [comment]
These are great notes and very helpful testing @KatieWoe @lmulhall-phet and others, thanks very much!

**Comment by jessegreenberg** [comment]
I am going to go ahead and close this issue but it will be referenced in future issues and discussions.

**Comment by jessegreenberg** [comment]
Actually, reopening to check on whether or not only updating positioning every few seconds produces a noticeable change on some of the platforms that were marked as very problematic. @KatieWoe or @lmulhall-phet could you please test this version of GAO (maybe with multiple planets moving):
https://phet-dev.colorado.edu/html/gravity-and-orbits/1.2.0-pdomTransform.2/phet/gravity-and-orbits_en_phet.html?accessibility&mobileA11yTest

And see if the performance is acceptable? If much better, do you see the sim skip every 5 seconds or so?
On
 - [ ] iOS10
 - [ ] Chrome OS
 - [ ] Windows 7 Chrome
 - [ ] Mac OS 10.14 Chrome

If the sim still crashes, no need to continue testing on any of the remaining in the list.

**Comment by KatieWoe** [comment]
Mac OS 10.14 Chrome performance is still abysmal and doesn't seem to have changed. @jessegreenberg

**Comment by KatieWoe** [comment]
@jessegreenberg Also seems to be no change on iOS 10.  Per instructions in https://github.com/phetsims/QA/issues/197#issuecomment-426097533 stopping testing at this point.

**Comment by jessegreenberg** [comment]
Thanks @KatieWoe, that is surprising, but very helpful to know. In that case, I will hit the drawing board and hopefully return with an updated version.

**Comment by jessegreenberg** [comment]
I am going to close this issue and open a new one to avoid a megathread.
